### PR TITLE
Fix use of incorrect execution state in the table executor.

### PR DESCRIPTION
**Pull request by fruffy**
Need to clone before executing default action.

### CODE REVIEW

**Comment by fruffy** [comment]
@RabbitWhite1 This includes two bug fixes. In one case we did not clone an execution state before execution other branches, which leads to an incorrect analysis. In the second case, we incorrectly computed some mux collapse operation.
 
These bug fixes do not seem to affect dead code but do dramatically affect performance. To keep the analysis reasonable I had to reenable some simplification when producing a mux and disable constant folding and strength reduction. 

I really do not have a concrete pulse on the performance tradeoffs here. The scion and dash programs seem to have different kinds of characteristics which require different optimizations.

**Comment by RabbitWhite1** [comment]
|program|cyclomatic complexity|parser paths|
|:-|:-|:-|
|scion|137|14913|
|dash-pipeline-v1model-bmv2|229|31|

Scion's parser is complicated, while dash program seems more complicated in controls. Can this be a reason that makes suc ha difference? I am trying to generate performance statistics with/without some optimization and see what characteristic plays a role here.

**Comment by fruffy** [comment]
> program 	cyclomatic complexity 	parser paths
> scion 	137 	14913
> dash-pipeline-v1model-bmv2 	229 	31
> 
> Scion's parser is complicated, while dash program seems more complicated in controls. Can this be a reason that makes suc ha difference? I am trying to generate performance statistics with/without some optimization and see what characteristic plays a role here.

Interesting, this could definitely play a role.

File: core/simplify_expression.cpp [hunk 1]
```diff
@@ -565,7 +581,7 @@ const IR::Expression *produceSimplifiedMux(const IR::Expression *cond,
                                            const IR::Expression *falseExpression) {
     Util::ScopedTimer timer("Mux optimization");
     auto *mux = new IR::Mux(trueExpression->type, cond, trueExpression, falseExpression);
-    return mux->apply(FoldMuxConditionDown());
+    return simplify(mux->apply(FoldMuxConditionDown()));
```
Comment (RabbitWhite1) [inline core/simplify_expression.cpp:584]: Can this be simply `return simplify(mux);` given FoldMuxConditionDown is one of the ExpressionRewriter passes?

Comment (fruffy) [reply core/simplify_expression.cpp:584]: Will fix that

**Pull request merged**